### PR TITLE
[DOXIA-749] Correctly indent and separate blocks inside list items

### PR DESCRIPTION
**Pull request by kwin**
Block elements should be surrounded by blank lines inside list items to be compliant with more MD parsers

### CODE REVIEW

File: doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownSink.java [hunk 1]
```diff
@@ -859,6 +873,9 @@ public void nonBreakingSpace() {
 
     @Override
     public void text(String text, SinkEventAttributes attributes) {
+        if (text.equals("\n")) {
```
Comment (michael-o) [inline doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownSink.java:876]: Is this portable POSIX/Windows?

Comment (kwin) [reply doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownSink.java:876]: Good catch, this is a leftover from some debugging, is not supposed to be in here.

Comment (kwin) [reply doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownSink.java]: It turned out that the HTML parser often emits `text("\n")`without any semantical meaning. It becomes semantical if prefixing that with spaces, though. Therefore added a workaround to not prefix empty line breaks.

File: doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownSinkTest.java [hunk 2]
```diff
@@ -513,4 +510,24 @@ public void testHeadingAfterInlineElement() {
...
     }
+
+    @Test
+    public void testMultilineVerbatimSourceAfterListItem() {
+        try (final Sink sink = getSink()) {
+            sink.list();
+            sink.listItem();
+            sink.text("Before");
+            sink.verbatim(SinkEventAttributeSet.SOURCE);
+            sink.text("codeline1\ncodeline2");
```
[the remark this answers is not in the document]
Comment (kwin) [reply doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownSinkTest.java]: This is supposed to be a multiline comment, `\n` by EOL now.

**Pull request merged**